ocaml/xapi/create_misc.ml

```diff
@@ -109,28 +112,12 @@ let read_dom0_memory_usage () =
   with _ ->
     None
 
-let read_localhost_info () =
-  let xen_verstring, total_memory_mib =
-    try
-      let xc = Xenctrl.interface_open () in
-      let v = Xenctrl.version xc in
-      Xenctrl.interface_close xc;
-      let open Xenctrl in
-      let xen_verstring = Printf.sprintf "%d.%d%s" v.major v.minor v.extra in
-      let total_memory_mib =
-        let open Xapi_xenops_queue in
-        let module Client = (val make_client (default_xenopsd ()) : XENOPS) in
-        Client.HOST.get_total_memory_mib "read_localhost_info" in
-      xen_verstring, total_memory_mib
-    with e ->
-      if Pool_role.is_unit_test ()
-      then "0.0.0", 0L
-      else begin
-        warn "Failed to read xen version";
-        match Balloon.get_memtotal () with
-        | None -> "unknown", 0L
-        | Some x -> "unknown", Int64.(div x (mul 1024L 1024L))
-      end
+let read_localhost_info ~__context =
+  let open Xapi_xenops_queue in
```

Comment:
Do we need this `open`?

Reply:
It it necessary to be able to use `XENOPS` as Xenops_client.Client, on the following line.

There might be a better way to use the Xenopsd API client, but this is how I saw it being used everywhere else.

Reply:
Otherwise is would be `Xapi_xenops_queue.XENOPS`. Let's keep iusing `open`.